cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/ClientConfig.java

```diff
@@ -53,6 +53,11 @@ public class ClientConfig
     public static final String SNAPSHOT_NAME_KEY = "snapshotName";
     public static final String DC_KEY = "dc";
     public static final String CREATE_SNAPSHOT_KEY = "createSnapshot";
+    /**
+     * Option to filter distinct instances before creating snapshots. This is only applicable when
+     * using vnodes where the token ring will contain multiple entries per instance.
+     */
+    public static final String CREATE_SNAPSHOT_FILTER_DISTINCT_INSTANCES_KEY = "createSnapshotFilterDistinctInstances";
```

Comment:
Meanwhile the spark option works, it would be nicer if we can figure it out _automatically_ if the cluster is running with vnode enabled, and remove the option. I think it is achievable. For example, when there are multiple tokens (vnode), there are entries with the same fqdn but different token.

Reply:
Though we're already assuming uniqueness of FQDN when this flag is set (which defaults to true). Would you prefer we set this value based on whether FQDNs are entirely unique in the ring structure?

Reply:
IMO, a flag is wanted, when (1) there is nothing in the system to derive a condition (that alters behavior) and it requires an external signal, or (2) explicitness is _required_. I think in this case, if we can reliably derive the condition from the ring response, it would void the need for the flag. Beside, it might cause confusion when cluster is running vnode but `createSnapshotFilterDistinctInstances` is set to `false`. I think the expectation is to only create snapshot once on each physical host always. If true, the flag is also unnecessary.

Reply:
> We could leverage the [NodeSettings#tokens](https://github.com/apache/cassandra-sidecar/blob/trunk/client-common/src/main/java/org/apache/cassandra/sidecar/common/response/NodeSettings.java#L48) to make the automatic determination.

@frankgh and I talked offline and I am re-posting the summary. Let's prefer deriving vnode from ring response. The problem of `nodesettings#tokens` is that it picks from a single sidecar instance. Cassandra does not enforce that all nodes in the cluster should configure vnodes. In other words, some nodes can be configured to run single token, but others with vnodes. If relying on the tokens count from a random single node, it could provide the misinformation.

Reply:
I'm in agreement here. The Sidecar will throw an error if we try to create the same snapshot multiple times on the same host so it makes sense for this to be the default logic. Regarding uniqueness, is FQDN sufficient or would we prefer FQDN+Port?

Reply:
> Regarding uniqueness, is FQDN sufficient or would we prefer FQDN+Port?

FQDN+Port combination is more reliable and it aligns with Cassandra configuration better, since one can run multiple Cassandra instance on the same host.

Reply:
Updated